Orange/widgets/visualize/tests/test_owpythagorastree.py

```diff
@@ -283,3 +284,46 @@ def test_label_on_tree_connect_and_disconnect(self):
         self.assertNotRegex(
             self.widget.info.text(), regex,
             'Initial info should not contain node or depth info')
+
+    def test_tree_determinism(self):
+        """Check that the tree is drawn identically upon receiving the same
+        dataset with no parameter changes."""
+        n_tries = 10
+
+        def _check_all_same(data):
+            """Check that all the elements within an iterable are identical."""
+            iterator = iter(data)
+            try:
+                first = next(iterator)
+            except StopIteration:
+                return True
+            return all(first == rest for rest in iterator)
+
+        # Make sure the tree are deterministic for iris
+        scene_nodes = []
+        for _ in range(n_tries):
+            self.send_signal(self.signal_name, self.signal_data)
+            scene_nodes.append([n.pos() for n in self.get_visible_squares()])
+        for node_row in zip(*scene_nodes):
+            self.assertTrue(
+                _check_all_same(node_row),
+                "The tree was not drawn identically in the %d times it was "
+                "sent to widget after receiving the iris dataset." % n_tries
+            )
+
+        # Make sure trees are deterministic with data where some variables have
+        # the same entropy
+        data_same_entropy = Table(path.join(
+            path.dirname(__file__), "../../tests/datasets/same_entropy.tab"))
```

Comment:
Is there a cleaner solution than `path.join(path.dirname(path.dirname(path.dirname(__file__))), "tests", "datasets", "same_entropy.tab")`? Could I maybe put in `".."` instead of  `path.dirname`?